draft-ietf-httpbis-digest-headers.md

```diff
@@ -358,152 +438,213 @@ Examples:
+~~~
+
+
+# Digest Algorithm Registry {#algorithms}
+
+The "HTTP Digest Algorithm Registry", maintained by IANA at
+<https://www.iana.org/assignments/http-dig-alg/> registers algorithms for use
+with a range of HTTP fields.
+
+Algorithms used in `Digest` or `Want-Digest` are communicated using the value of
+`Digest-algorithm token`. `Digest` also includes an encoded output, the format
+is specified by the value of `Digest-algorithm output format`.
+
+Algorithms used in `Representation-Digest` and `Content-Digest` are communicated
+using the value of `Digest Dictionary key`. The encoding for these fields is
+always sf-binary.
+
+Algorithms used in `Want-Representation-Digest` and `Want-Content-Digest` are
+communicated using the value of `Digest List item`.
 
-The "HTTP Digest Algorithm Values Registry",
-maintained by IANA at <https://www.iana.org/assignments/http-dig-alg/> registers
-digest-algorithm values.
 Registrations MUST include the following fields:
 
- - Digest algorithm: the token value.
-     The registry can be used to reserve token values
+ - Algorithm: a friendly name of the algorithm
+ - Description: a short description of the algorithm
+ - Digest-algorithm token: the token value used in `Digest` and `Want-Digest`
+   fields
+ - Digest-algorithm output format: the encoding format of output used in `Digest`
+ - Digest Dictionary key: the Structured Fields key value used in
+   `Representation-Digest` or `Content-Digest` field Dictionary member keys
+ - Digest List item: the Structured Fields item used as
+   `Want-Representation-Digest` or `Want-Content-Digest` field List members
  - Status: the status of the algorithm.
      Use "standard" for standardized algorithms without known problems;
      "experimental" or some other appropriate value
      - e.g. according to the type and status of the primary document
      in which the algorithm is defined;
      "insecure" when the algorithm is insecure;
      "reserved" when the algorithm references a reserved token value
- - Description: the description of the digest-algorithm and its encoding
- - Reference: a set of pointers to the primary documents defining the digest-algorithm
-
-The associated encoding for new digest-algorithms MUST either
-be represented as a quoted string
-or MUST NOT include ";" or "," in the character sets used for the encoding.
+ - Reference: a set of pointers to the primary documents defining the algorithm
+   and key
 
 Insecure digest algorithms MAY be used to preserve integrity against corruption, but MUST
-NOT be used in a potentially adversarial setting; for example, when signing digest fields' values for
+NOT be used in a potentially adversarial setting; for example, when signing integrity fields' values for
 authenticity.
 
-The registry is initialized with the tokens listed below.
+The registry is initialized with the algorithms listed below.
 
   {: vspace="0"}
-  sha-512
-  : * Digest Algorithm: sha-512
+  SHA-512
```

Comment:
why uppercase?

Reply:
My idea here is to modify the table to be rows of Algorithms, not rows of tokens. The encoding rules for legacy digest and SF has the potential for incompatibility. It's feasible that somebody has defined a `token` that cannot be encoded as a `key` - so we have multiple columns in the table that spells this out explicitly.

I'm open to other options here but we do need to consider a robust solution given our constraints. This is robust if a bit ugly.

To also answer you question below, T

Reply:
> It's feasible that somebody has defined a token that cannot be encoded as a key

if such tokens do not exist yet, I will just restrict the space of possible values for all fields.

Reply:
Since there is no requirement to register algorithms before usage, I'm not sure how enforceable changing things is. 

`token` has a definition and trying to change that charset now in relation to digest seems like a backwards compatibility breaking change.